### PR TITLE
Grades modal

### LINKED ISSUE #340: Grades Modal
- Create a drop down grades graph. This can be a simple bar graph. 
- Rename the button from "Zotistics" to "Past Grades"
- At the bottom of the bar graph, show a button that says "View on Zotistics" that links to more info.

Here is an example of what ZotCourse does: 
image

It's up to you how much you want to implement inside the bar graph. I'd say start with just the average from the course and then later we can add more specific bars like Instructor and last offering.

### PR DESCRIPTION
**Pull request by kartik-bhagatwala**
## Summary

Create a grades modal to show a basic graph of past grades data for the class, and replaces the Zotistics button. Data is pulled from peterportal API. The modal also includes a button to view the class on Zotistics, in line with current functionality.
image
image

## Test Plan
Tested with different classes, verified that classes with no valid response from Peterportal handled the absence of data gracefully, and displayed an appropriate error message.

## Issues
Closes #340 

## Future Followup  
Implement teacher grade data. If theres only one teacher, then calculate percentage distribution and plot the individual teacher and the overall average for the class on the chart. If there are multiple teachers, then plot bars for all of them on the chart so users can easily compare them.

### CODE REVIEW

File: src/components/SectionTable/GradesPopup.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                            <Bar dataKey="all" fill="#5182ed" />
+                        </BarChart>
+                    </ResponsiveContainer>
+                )}
+                <div style={{ marginBottom: '5px', textAlign: 'center' }}>
+                    <CourseInfoButton
+                        text="Zotistics"
+                        redirectLink={`https://zotistics.com/?&selectQuarter=&selectYear=&selectDep=${encodedDept}&classNum=${courseNumber}&code=&submit=Submit`}
+                    />
+                </div>
```
Comment (ChaseC99) [inline src/components/SectionTable/GradesPopup.js]: This should be an `<a>` tag, not a courseInfoButton

File: src/components/SectionTable/GradesPopup.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                <div className={classes.gpaTitle}>{graphTitle}</div>
+                {gradeData && (
+                    <ResponsiveContainer width={width} height={height}>
+                        <BarChart data={gradeData}>
+                            <CartesianGrid strokeDasharray="3 3" />
+                            <XAxis dataKey="name" tick={{ fontSize: 12 }} />
+                            <YAxis tick={{ fontSize: 12 }} width={40} />
+                            <Bar dataKey="all" fill="#5182ed" />
+                        </BarChart>
+                    </ResponsiveContainer>
```
Comment (ChaseC99) [inline src/components/SectionTable/GradesPopup.js]: Axis labels should be white in dark mode. It's hard to see the gray.

**Review by ChaseC99 — CHANGES_REQUESTED** [review]
This is looking solid!

I've done some refactoring (see previous commits) but there are still a few comments to address. 

Once those are done it should be good to ship!

File: src/components/SectionTable/GradesPopup.js (hunk 1 above)
Comment (ChaseC99) [reply src/components/SectionTable/GradesPopup.js]: Completed. Styling can be done as a fellowship task.
image

File: src/components/SectionTable/GradesPopup.js (hunk 2 above)
Comment (kartik-bhagatwala) [reply src/components/SectionTable/GradesPopup.js]: Thanks for all the work on the refactor! Code looks a lot cleaner.

changed the axis labels to be white for dark mode with ef68c73.

image

**Review by ChaseC99 — APPROVED** [review]
Looks epic! Great work on this feature 🎉

**Pull request merged**